nntrainer/optimizers/sgd.h

```diff
@@ -39,10 +38,12 @@ class SGD : public Optimizer {
                       int iteration);
 
   /**
-   * @brief     get the base name for the optimizer
-   * @retval    base name of the optimizer
+   * @copydoc Optimizer::getType()
    */
-  std::string getBaseName() { return "SGD"; };
+  const std::string getType() const { return SGD::type; }
```

Comment:
ditto - move to optimizer_internal.h

Reply:
Optimizer.h has getType(). Could I get the reason for optimizer_internal to have this?

Reply:
I missed that optimizer.h has getType().